libs/runtime/ebpf_hash_table.c

```diff
@@ -340,6 +340,33 @@ _ebpf_hash_table_bucket_entry(size_t key_size, _In_ const ebpf_hash_bucket_heade
     return (ebpf_hash_bucket_entry_t*)(offset + (size_t)index * entry_size);
 }
 
+/**
+ * @brief Helper function to ensure correct memory ordering when reading a bucket from the hash table.
+ *
+ * @param[in] hash_table Pointer to the hash table.
+ * @param[in] bucket_index Index of the bucket to read.
+ * @return Pointer to the bucket or NULL if the bucket is empty.
+ */
+static inline ebpf_hash_bucket_header_t*
+_ebpf_hash_table_get_bucket(_In_ const ebpf_hash_table_t* hash_table, size_t bucket_index)
+{
+    return (ebpf_hash_bucket_header_t*)ReadSizeTAcquire((ULONG_PTR*)&(hash_table->buckets[bucket_index].header));
```

Comment:
I'm still not 100% sure this is necessary (and I'm absolutely not 100% sure it's _un_-necessary) but since it has shown minimal perf cost on arm64 and obviously no cost on x64, being conservative seems good.

Reply:
Based on discussions with various processor experts, the answer seems to be ambiguous. Some say it's needed other say it's not. Best to err on the side of caution in my opinion.